repository-commons/src/main/java/io/aiven/elasticsearch/repositories/io/CryptoIOProvider.java

```diff
@@ -49,40 +45,24 @@ public CryptoIOProvider(final SecretKey encryptionKey) {
         this.encryptionKey = encryptionKey;
     }
 
-    public void compressAndEncrypt(final InputStream chunk,
-                                   final WritableByteChannel writer) throws IOException {
-        final var out = Channels.newOutputStream(writer);
-        Streams.copy(
-                chunk,
-                new ZstdOutputStream(
-                        new CipherOutputStream(
-                                out,
-                                createEncryptingCipherAndWriteIv(out)
-                        )
-                )
-        );
-    }
-
-    private Cipher createEncryptingCipherAndWriteIv(final OutputStream out) throws IOException {
+    public long encrypt(final InputStream in,
```

Comment:
`compressAndEncrypt` seems still valid and more accurate. Any reason to rename?

Reply:
True, it was an idea, which wont work. I will return name back